tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -116,18 +115,22 @@ def bulk_create_records(unsaved_records, line_number, header_count, datafile, df
                 num_db_records_created += len(created_objs)
                 num_elastic_records_created += document.update(created_objs)[0]
             except ElasticsearchException as e:
-                logger.error(f"Encountered error while indexing datafile documents: {e}")
-                LogEntry.objects.log_action(
-                    user_id=datafile.user.pk,
-                    content_type_id=ContentType.objects.get_for_model(DataFile).pk,
-                    object_id=datafile,
-                    object_repr=f"Datafile id: {datafile.pk}; year: {datafile.year}, quarter: {datafile.quarter}",
-                    action_flag=ADDITION,
-                    change_message=f"Encountered error while indexing datafile documents: {e}",
-                )
+                log_parser_exception(datafile,
+                                     f"Encountered error while indexing datafile documents: \n{e}",
+                                     "error"
+                                     )
                 continue
+            except DatabaseError as e:
+                log_parser_exception(datafile,
+                                     f"Encountered error while creating database records: \n{e}",
+                                     "error"
+                                     )
+                return False
             except Exception as e:
-                logger.error(f"Encountered error while creating datafile records: {e}")
+                log_parser_exception(datafile,
+                                     f"Encountered generic exception while creating database records: \n{e}",
```

Comment:
I was able to trigger this exception, but it doesn't generate an exception-related email notification.